blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/NodeDownstreamBuildAction.java

```diff
@@ -0,0 +1,54 @@
+package io.jenkins.blueocean.service.embedded.rest;
+
+import hudson.model.InvisibleAction;
+import hudson.model.Run;
+import org.jenkinsci.plugins.workflow.actions.FlowNodeAction;
+import org.jenkinsci.plugins.workflow.graph.FlowNode;
+
+import java.util.Objects;
+
+/**
+ * Annotates a FlowNode to point to a downstream build triggered by said node. Applied by
+ * io.jenkins.blueocean.listeners.DownstreamJobListener in blueocean-pipeline-api-impl
+ */
+public class NodeDownstreamBuildAction extends InvisibleAction implements FlowNodeAction, BluePipelineAction {
+
+    private final String runExternalizableId;
+    private final String description;
+
+    public NodeDownstreamBuildAction(String runExternalizableId, String description) {
```

Comment:
It should take link reference to run. This way, it will be self contained and won't need patching link later on.